dashmips/syscalls/print_syscalls.py

```diff
@@ -1,42 +1,56 @@
 """Syscalls related to printing."""
 from . import mips_syscall
 from ..models import MipsProgram
-from ..utils import intify
+from ..utils import intify, bytesify
+from struct import Struct
 
 
 @mips_syscall(4)
 def print_string(program: MipsProgram):
-    """Print string at address provided in $a0."""
+    """Print string. $a0 = address of null-terminated string to print."""
     address = program.registers["$a0"]
     string = program.memory.read_str(address)
-    print(string, end="")
+    print(string, end=" ")
 
 
 @mips_syscall(11)
 def print_char(program: MipsProgram):
-    """Print string at address provided in $a0."""
+    """Print character. $a0 = character to print."""
     character = chr(program.registers["$a0"] & 0xFF)
     if ord(" ") <= ord(character) <= ord("~") or character in ("\n", "\t", "\r"):
         # is printable
-        print(character, end="")
+        print(character, end=" ")
 
 
 @mips_syscall(1)
 def print_int(program: MipsProgram):
-    """Print Int in Dec."""
-    print(program.registers["$a0"], end="")
+    """Print Int in Dec. $a0 = integer to print."""
+    print(program.registers["$a0"], end=" ")
 
 
 @mips_syscall(34)
 def print_hex(program: MipsProgram):
-    """Print Int in Hex."""
-    print(f"{program.registers['$a0'] & 0xFFFFFFFF:08x}", end="")
+    """Print Int in Hex. $a0 = integer to print."""
+    print(f"0x{program.registers['$a0'] & 0xFFFFFFFF:08x}", end=" ")
 
 
 @mips_syscall(35)
 def print_bin(program: MipsProgram):
-    """Print Int in Bin."""
-    print(f"{program.registers['$a0'] & 0xFFFFFFFF:032b}", end="")
+    """Print Int in Binary. $a0 = integer to print."""
+    print(f"{program.registers['$a0'] & 0xFFFFFFFF:032b}", end=" ")
```

Comment:
https://github.com/nbbeeken/dashmips/pull/15/files#r458161189
This still needs updating

Reply:
Huh, swore I made this change.  Should be good now!